projects/batfish/src/main/java/org/batfish/job/BatfishJobExecutor.java

```diff
@@ -41,105 +57,162 @@ public BatfishJobExecutor(
     }
   }
+
+  ExecutorService createExecutorService() {
+    ExecutorService pool;
+    if (_settings.getSequential()) {
+      return Executors.newSingleThreadExecutor();
+    } else {
+      int maxConcurrentThreads = Runtime.getRuntime().availableProcessors();
+      int numConcurrentThreads = Math.min(maxConcurrentThreads, _settings.getJobs());
+      return Executors.newFixedThreadPool(numConcurrentThreads);
+    }
+  }
+
+  String getFailureMessage(JobResultT result) {
+    String time = CommonUtil.getTime(result.getElapsedTime());
+    String failureMessage =
+        String.format(
+            "Failure running job after elapsed time: %s\n-----"
+                + "BEGIN JOB LOG-----\n%s\n-----END JOB LOG-----",
+            time, result.getHistory().toString(BatfishLogger.getLogLevel(_settings.getLogLevel())));
+    return failureMessage;
+  }
+
+  String getSuccessMessage(JobResultT result) {
+    String time = CommonUtil.getTime(result.getElapsedTime());
+    String successMessage =
+        String.format(
+            "Job terminated successfully with result: %s after elapsed time: %s - %d/%d "
+                + "(%.1f%%) complete\n",
+            result, time, _finishedJobs, _totalJobs, _finishedPercent);
+    return successMessage;
+  }
+
+  /**
+   * Handles the Job result appropriately in case of success/error, and updates logs
+   *
+   * @param result Result of Job
+   * @param output Data structure to which the result of the job will be applied, it should be of
+   *     generic type {@link OutputT}
+   * @param answerElement {@link AnswerElement} for the execution of jobs
+   * @param failureCauses List of failure causes for all jobs
+   * @return {@code true} if there is a processing error
+   */
+  boolean handleJobResult(
+      JobResultT result,
+      OutputT output,
+      AnswerElementT answerElement,
+      List<BatfishException> failureCauses) {
+    Throwable failureCause = result.getFailureCause();
+    if (failureCause == null) {
+      result.applyTo(output, _logger, answerElement);
+      _logger.info(getSuccessMessage(result));
+      return false;
+    } else {
+      String failureMessage = getFailureMessage(result);
+      BatfishException bfc = new BatfishException(failureMessage, failureCause);
+      if (_settings.getExitOnFirstError()) {
+        result.appendHistory(_logger);
+        throw bfc;
+      } else {
+        result.appendHistory(_logger);
+        _logger.errorf("%s:\n\t%s", failureMessage, ExceptionUtils.getStackTrace(failureCause));
+        failureCauses.add(bfc);
+        if (!_haltOnProcessingError) {
+          result.applyTo(output, _logger, answerElement);
+        }
+        return true;
+      }
+    }
+  }
+
+  void handleProcessingError(List<JobT> jobs, List<BatfishException> failureCauses) {
+    int numJobs = jobs.size();
+    int numFailed = failureCauses.size();
+    int numSucceeded = numJobs - numFailed;
+    if (!_logger.isActive(BatfishLogger.LEVEL_INFO)) {
```

Comment:
I don't understand this block -- we check if LEVEL_INFO is *disabled* and then, if it's disabled, log at INFO level.

What's going on?